### ISSUE TITLE
[css-anchor-position] Should scroll-margin/padding have an effect on position-visibility: anchors-visible?

### ISSUE THREAD

**Issue by kizu** [w3c/csswg-drafts#10410]
Was playing with the `position-visibility: anchors-visible` (https://drafts.csswg.org/css-anchor-position-1/#position-visibility), and I think it might be useful to take the scrollable area's `scroll-padding` or the anchor's `scroll-margin` into consideration (https://drafts.csswg.org/css-scroll-snap/#propdef-scroll-padding). An example: https://codepen.io/kizu/pen/BaedPEB

https://github.com/w3c/csswg-drafts/assets/177485/378c29d2-6785-477a-8f2f-13f5be9f265c

Here we can see how if we have some fixed element reducing the effective visible area of our scrollable container, we'd want to consider it for the `position-visibility`. Hooking it onto `scroll-margin` and `scroll-padding` sounds logical, as this is very similar to other similar cases like scrolling or snapping to the targeted element.

**Comment by johannesodland** [comment]
Yeah, that would be useful in a lot of cases. 

It might be smart to do something similar to Scroll-driven Animations, where `view-timeline-inset: auto` is specced to use the values of `scroll-padding` while it is still possible to specify a custom length-percentage value.

**Comment by fantasai** [comment]
I think I'd like to re-open this issue. For things that are actually hidden this makes sense, but as long as the anchor is actually visible, we shouldn't consider it hidden. I think it would be surprising to users if a panel disappeared at some arbitrary point in its scroll passage across the screen.

The `scroll-padding` property has several uses, one of which is to provide some breathing room when snapping items into the viewport, and another of which which is to screen out areas where fixedpos toolbars are blocking content. Although for the first use case, we wouldn't want to take it into account, given the second use case, it could be reasonable. Maybe. But I don't think we should ever consider `scroll-margin` here.

**Comment by tabatkins** [comment]
Yeah, I can go either way. I agree that this *is* meant to handle the case when the anchor is fully hidden, so it seems somewhat reasonable to ignore scroll-margin/padding, but I'm fine either way (as expressed on the call).

**Comment by kizu** [comment]
Can this be controlled in some way? Very often sticky headers make the anchors in the main content fully hidden, and not always `scroll-padding` is possible to implement. If something can benefit from both use cases (using `scroll-margin` for getting this extra _margin_ from the ), I'd always want a way to configure it (not care too much about the default).

That said, maybe going with “`scroll-padding` has an effect, but `scroll-margin` is not” could be fine, as it will at least allow some control, but in practice I often see those used interchangeably.

**Comment by tabatkins** [comment]
Hmm, I could see that. However, we *really* just want the ability to declare that some element (like a sticky header) is an "obscuring" element, so that `scroll-padding:auto` can take that into account automatically based on the element's actual size/position. Right now, you have to guess at the size, which isn't great, and which is confusable with the *other* use-case, which is just providing a "comfortable margin" for the scrollable/pageable area (nothing to do with obscuring).

So I think we might want to (a) revert the resolution in this thread making us pay attention to scroll-padding/margin (so we'll go back to ignoring it), and (b) tag a new issue for scroll-snap-2/anchor-pos-2 about explicitly handling obscuring elements.

**Comment by kizu** [comment]
Hmm, true. If we were able to say that some element is “obscuring”, it could solve this, and some other cases pretty neatly, indeed.

Also, randomly remembered another case where using some element instead of indirectly using scroll-margins/paddings could help: https://github.com/w3c/csswg-drafts/issues/8905 (anchored sticky positioning; not related to this issue, but related to both anchor positioning and scroll-margins)

**Comment by fantasai** [comment]
Filed https://github.com/w3c/csswg-drafts/issues/12943 for follow-up. Proposal for this issue is, `position-visibility` does not pay attention to `scroll-padding` or `scroll-margin`.
